resources/assets/js/Pages/Calendar.js

```diff
@@ -59,35 +60,41 @@ export default class Calendar extends Component {
   }
 
   render() {
-    return (
-      [this.renderDays(),
-      this.state.start ? <div key="calendar-grid" className="calendar-grid">{this.renderWeeks()}</div> : '',
-      <div key="calendar-manage" className="calendar-manage">
-        {
-          this.props.isTutor ?
-          <div className="calendar-setting" onClick={(_)=>this.setState({ addLesson: true })}>
-            <img src="/images/icons8-plus-50.png" /> Add Lesson
+    return ([
+      <Sidebar selected={this.props.page} isTutor={this.props.isTutor} />,
+      <div className="width-fill flex-rows">
+        <div className="calendar-container">
+          {this.renderDays()}
+          {this.state.start ? <div key="calendar-grid" className="calendar-grid">{this.renderWeeks()}</div> : ''}
+          <div key="calendar-manage" className="calendar-manage">
+            {
+              this.props.isTutor ?
+              <div className="calendar-setting" onClick={(_)=>this.setState({ addLesson: true })}>
+                <img src="/images/icons8-plus-50.png" /> Add Lesson
+              </div>
+              : <div className="calendar-setting"></div>
+            }
+            <div className="calendar-setting">
+              <img src="/images/icons8-sort-down-filled-50.png" onClick={(_)=>this.nextWeek()} />
+              <img src="/images/icons8-sort-up-filled-50.png" onClick={(_)=>this.prevWeek()} />
+            </div>
           </div>
-          : <div className="calendar-setting"></div>
-        }
-        <div className="calendar-setting">
-          <img src="/images/icons8-sort-down-filled-50.png" onClick={(_)=>this.nextWeek()} />
-          <img src="/images/icons8-sort-up-filled-50.png" onClick={(_)=>this.prevWeek()} />
+          {this.state.addLesson ?
+            <ModalAddLesson
+              key="modal-add-lesson"
+              contacts={this.state.contacts}
+              subjects={this.state.subjects}
+              onAddLesson={()=>this.loadData(this.props)}
+              onCancel={()=>this.setState({ addLesson: false })}
```

Comment:
again you can surely create a function and pass in a boolean value to set the state. just an idea.